### PR TITLE
feat(cdn): add new data_source cdn domain statistics

### PR DESCRIPTION
**Pull request by Jason-Zhang9309**
**What this PR does / why we need it**:

add new data_source cdn domain statistics

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
add new data_source cdn domain statistics
```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/cdn' TESTARGS='-run TestAccDatasourceStatistics_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/cdn -v -run TestAccDatasourceStatistics_basic -timeout 360m -parallel 4
=== RUN   TestAccDatasourceStatistics_basic
=== PAUSE TestAccDatasourceStatistics_basic
=== CONT  TestAccDatasourceStatistics_basic
--- PASS: TestAccDatasourceStatistics_basic (12.62s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/cdn       12.758s
```

### CODE REVIEW

File: docs/data-sources/cdn_domain_statistics.md [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+  For HTTP status code statistics (one or more types can be returned), the value can be:
+  **http_code_2xx**: status codes 2xx.
+  **http_code_3xx**: status codes 3xx.
+  **http_code_4xx**: status codes 4xx.
+  **http_code_5xx**: status codes 5xx.
+  **status_code_2xx**: details of status code 2xx.
+  **status_code_3xx**: details of status code 3xx.
+  **status_code_4xx**: details of status code 4xx.
+  **status_code_5xx**: details of status code 5xx.
```
Comment (Lance52259) [inline docs/data-sources/cdn_domain_statistics.md]: missing list character `+`.

File: docs/data-sources/cdn_domain_statistics.md [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+    for example, 1631243700000, which means 2021-09-10 11:15:00.
+  + If the value of interval is 3600, set this parameter to a multiple of 1 hour,
+    for example, 1631325600000, which means 2021-09-11 10:00:00.
+  + If the value of interval is 86400, set this parameter to 00:00:00 (GMT+08:00),
+    for example, 1631376000000, which means 2021-09-12 00:00:00.
+
+* `interval` - (Optional, String) Specifies the query time interval, in seconds, the value can be,
+  **300**(5 minutes): Maximum query span 2 days
+  **3600**(1 hour): Maximum query span 7 days
+  **86400**(1 day): Maximum query span 31 days
```
Comment (Lance52259) [inline docs/data-sources/cdn_domain_statistics.md]: using list characters to show these enum values?
```
* `interval` - (Optional, String) Specifies the query time interval, in seconds, the value can be:
  +**300**(5 minutes): Maximum query span 2 days
  +**3600**(1 hour): Maximum query span 7 days
  +**86400**(1 day): Maximum query span 31 days
```

File: docs/data-sources/cdn_domain_statistics.md [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+* `end_time` - (Required,, String) Specifies the end timestamp of the query.
+  The timestamp must be set to a multiple of 5 minutes.
+  + If the value of interval is 300, set this parameter to a multiple of 5 minutes,
+    for example, 1631243700000, which means 2021-09-10 11:15:00.
+  + If the value of interval is 3600, set this parameter to a multiple of 1 hour,
+    for example, 1631325600000, which means 2021-09-11 10:00:00.
+  + If the value of interval is 86400, set this parameter to 00:00:00 (GMT+08:00),
+    for example, 1631376000000, which means 2021-09-12 00:00:00.
+
+* `interval` - (Optional, String) Specifies the query time interval, in seconds, the value can be,
```
Comment (Lance52259) [inline docs/data-sources/cdn_domain_statistics.md]: incorrect type.

File: docs/data-sources/cdn_domain_statistics.md (hunk 1 above)
Comment (Jason-Zhang9309) [reply docs/data-sources/cdn_domain_statistics.md]: fixed

File: docs/data-sources/cdn_domain_statistics.md (hunk 2 above)
Comment (Jason-Zhang9309) [reply docs/data-sources/cdn_domain_statistics.md]: fixed

File: docs/data-sources/cdn_domain_statistics.md (hunk 3 above)
Comment (Jason-Zhang9309) [reply docs/data-sources/cdn_domain_statistics.md]: fixed

File: huaweicloud/services/acceptance/cdn/data_source_huaweicloud_cdn_domain_statistics_test.go [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:          func() { acceptance.TestAccPreCheck(t) },
+		ProviderFactories: acceptance.TestAccProviderFactories,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccDatasourceStatistics_basic(),
+				Check: resource.ComposeTestCheckFunc(
+					dc.CheckResourceExists(),
+					resource.TestCheckResourceAttr(rName, "result", "{}\n"),
```
Comment (ShiChangkuo) [inline huaweicloud/services/acceptance/cdn/data_source_huaweicloud_cdn_domain_statistics_test.go]: if the value of result includes "\n", it better to trim it.

File: docs/data-sources/cdn_domain_statistics.md [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `domain_name` - (Required, String) Specifies the domain name list. Domain names are separated by commas (,),
+for example, www.test1.com,www.test2.com. The value all indicates that all domain names under your account are queried.
+
+* `action` - (Required, String) Specifies the action name. Possible values are: **location_summary** and **location_detail**.
```
Comment (ShiChangkuo) [inline docs/data-sources/cdn_domain_statistics.md:30]: we can genetate the validation func with **enum** in yaml

Comment (Jason-Zhang9309) [reply docs/data-sources/cdn_domain_statistics.md:30]: fixed

**Comment by Jason-Zhang9309** [comment]
```
make testacc TEST='./huaweicloud/services/acceptance/cdn' TESTARGS='-run TestAccDatasourceStatistics_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/cdn -v -run TestAccDatasourceStatistics_basic -timeout 360m -parallel 4
=== RUN   TestAccDatasourceStatistics_basic
=== PAUSE TestAccDatasourceStatistics_basic
=== CONT  TestAccDatasourceStatistics_basic
--- PASS: TestAccDatasourceStatistics_basic (11.19s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/cdn       11.276s
```

**Pull request merged**